### PR TITLE
Adding list view pattern

### PR DESCRIPTION
**Pull request by cshinn**
Adds the list view pattern as it currently exists on patternfly.org

### CODE REVIEW

File: pattern-library/content-views/list-view/design/design.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# List View
+
+## Standard List View
+![List view](img/list-view-callout.png)
+1. **Data Toolbar:** The Data Toolbar pattern may include filter, sort, actions, find and change view type.
+1. **Row:**
```
Comment (LHinson) [inline pattern-library/content-views/list-view/design/design.md]: Formatting issue. "Row" should be #2 and on a new line.

File: pattern-library/content-views/list-view/design/design.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  - If drill down behavior is supported, clicking on an item will link to the appropriate page.
+  - Each item in List View should have the same layout
+    - When including titles, Make the title bold rather than appending a colon ( : )
+    - inks may be displayed within an item
+  - When the viewport size is decreased:
+    - Content should wrap, displaying multiple rows of data
+    - Content could be eliminated
+1. **Multi-Selection via Checkboxes** (optional): If multi-selection is supported, checkboxes appear on the left. Actions in the Data Toolbar apply to the selected items.
```
Comment (LHinson) [inline pattern-library/content-views/list-view/design/design.md]: Formatting issue. This should be number 3

File: pattern-library/content-views/list-view/design/design.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+1. **Item Hover:** A hover state should be shown visually when a user hovers over an item.
```
Comment (LHinson) [inline pattern-library/content-views/list-view/design/design.md]: this should be number 4

File: pattern-library/content-views/list-view/design/design.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+1. **Inline Actions:** Inline actions should be included if the user needs to be able to quickly perform an action on a single item.
```
Comment (LHinson) [inline pattern-library/content-views/list-view/design/design.md]: this should be 5

File: pattern-library/content-views/list-view/design/overview.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+A List View displays data in rows. Each row displays the same set of attributes, although not necessarily displayed in columns, and the attributes may wrap. The List View omits column headers and other visual noise, making it a cleaner visualization than a data table.
+
+Jump to [Standard List View](https://www.patternfly.org/pattern-library/content-views/list-view/#/standard), [List View with Expanding Rows](https://www.patternfly.org/pattern-library/content-views/list-view/#/expanding-rows) or [Row Variation Examples](https://www.patternfly.org/pattern-library/content-views/list-view/#/row-variations)
+
+## Standard List View
+
+[View full page example](https://www.patternfly.org/pattern-library/content-views/list-view/list-view.html#_)
+
+## List View with Expanding Rows
```
Comment (LHinson) [inline pattern-library/content-views/list-view/design/overview.md]: There should be an image shown above this link

File: pattern-library/content-views/list-view/design/overview.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+![List view with an expanded row](img/list-view-simple-expansion.png)
+
+### Compound Expansion
+
+Compound expansion supports multiple expansion panels to examine multiple attributes of the selected row.
+
+![List view with compound row expansion](img/list-view-compound-expansion-example.png)
+
+## Row Variation Examples
```
Comment (LHinson) [inline pattern-library/content-views/list-view/design/overview.md]: There should be images of the row variations

**Comment by LHinson** [comment]
Comments inline @cshinn

**Comment by cshinn** [comment]
@LHinson I've updated the number formatting in the design file. With respect to the overview file, The missing elements were live HTML examples on the site. Should I put in some sort of placeholder text, or should I take a screenshot and put it in? or something else?

**Comment by LHinson** [comment]
@cshinn take a screenshot and place it in. Thanks :)

File: pattern-library/content-views/list-view/design/design.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
+# List View
+
+## Standard List View
+
+![List view](img/list-view-callout.png)
+
+1. **Data Toolbar:** The Data Toolbar pattern may include filter, sort, actions, find and change view type.
```
Comment (LHinson) [inline pattern-library/content-views/list-view/design/design.md]: This should just say "Toolbar" not "Data Toolbar". It should also link to the toolbar page.

**Comment by LHinson** [comment]
Nice. Looks great @cshinn 

@mcarrano @andresgalante would be great if one of you could also review.

**Comment by mcarrano** [comment]
@LHinson @cshinn I just skimmed this, but looks good to me assuming that content is basically unchanged from current pattern design.

**Comment by catrobson** [comment]
+1 - looks good to me

**Pull request merged**